napari/_qt/qt_plugin_list.py

```diff
@@ -0,0 +1,301 @@
+"""Provides a QtPluginSorter that allows the user to change plugin call order.
+"""
+from typing import Optional
+
+from pluggy.manager import HookImpl, PluginManager, _HookCaller
+from qtpy.QtCore import QEvent, Qt, Signal, Slot
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QComboBox,
+    QDialog,
+    QFrame,
+    QGraphicsOpacityEffect,
+    QHBoxLayout,
+    QLabel,
+    QListWidget,
+    QListWidgetItem,
+    QSizePolicy,
+    QVBoxLayout,
+    QWidget,
+)
+
+from .utils import drag_with_pixmap
+from ..plugins import plugin_manager as napari_plugin_manager
+from ..plugins.utils import HookOrderType, permute_hook_implementations
+
+
+class ImplementationListItem(QFrame):
+    """A Widget to render each hook implementation in a QtHookImplListWidget.
+
+    Parameters
+    ----------
+    item : QListWidgetItem
+        An item instance from a QListWidget. This will most likely come from
+        :meth:`QtHookImplListWidget.add_hook_implementation_to_list`.
+    parent : QWidget, optional
+        The parent widget, by default None
+
+    Attributes
+    ----------
+    plugin_name_label : QLabel
+        The name of the plugin providing the hook implementation.
+    enabled_checkbox : QCheckBox
+        Checkbox to set the ``enabled`` status of the corresponding hook
+        implementation.
+    opacity : QGraphicsOpacityEffect
+        The opacity of the whole widget.  When self.enabled_checkbox is
+        unchecked, the opacity of the item is decreased.
+    """
+
+    def __init__(self, item: QListWidgetItem, parent: QWidget = None) -> None:
+        super().__init__(parent)
+        self.setToolTip("Click and drag to change call order")
+        self.item = item
+        self.opacity = QGraphicsOpacityEffect(self)
+        self.setGraphicsEffect(self.opacity)
+        layout = QHBoxLayout()
+        self.setLayout(layout)
+
+        self.position_label = QLabel()
+        self.update_position_label()
+
+        self.plugin_name_label = QLabel(item.hookimpl.plugin_name)
+        self.enabled_checkbox = QCheckBox(self)
+        self.enabled_checkbox.setToolTip("Uncheck to disable this plugin")
+        self.enabled_checkbox.stateChanged.connect(self._set_enabled)
+        self.enabled_checkbox.setChecked(
+            getattr(item.hookimpl, 'enabled', True)
+        )
+        layout.addWidget(self.position_label)
+        layout.addWidget(self.enabled_checkbox)
+        layout.addWidget(self.plugin_name_label)
+        layout.setStretch(2, 1)
+        layout.setContentsMargins(0, 0, 0, 0)
+
+    def _set_enabled(self, state: bool) -> None:
+        """Set the enabled state of this hook implementation to ``state``."""
+        # "hookimpl.enabled" is NOT a pluggy attribute... we are adding that to
+        # allow skipping of hook_implementations.
+        # see plugins.io.read_data_with_plugins() for an example
+        setattr(self.item.hookimpl, 'enabled', bool(state))
```

Comment:
I'm not sure why I used `setattr` here.  brain fart?  state needs to be cast because the checkbox.stateChanged signal may return an int.  I'll update the type signature